WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadHandler.java

```diff
@@ -286,7 +286,7 @@ protected UploadPostTaskResult doInBackground(PostModel... posts) {
                     break;
                 case REMOTE_AUTO_SAVE:
                     AppLog.d(T.POSTS, "PostUploadHandler - REMOTE_AUTO_SAVE. Post: " + mPost.getTitle());
-                    mDispatcher.dispatch(PostActionBuilder.newRemoteAutoSavePostAction(payload));
+                    mAutoSavePostIfNotDraftUseCase.autoSavePostOrUpdateDraft(payload, PostUploadHandler.this);
```

Comment:
> Also what happens if the fetchPostStatus request fails? The current implementation still pushes the post, but I assume this is not what we'll want to do. Ideally we'd return what happened from the "doInBackground", right?

That's only because it's not yet implemented. I think we should figure out how we want the use case to behave first and then decide what to do with the result so we don't have to change it later.

---

For the rest of this comment, I am.. kind of confused. I re-read the comment multiple times, but I don't understand the worry about the `PUSH_POST_DISPATCHED` probably because I don't fully understand what's happening when we push post.

One very important reason for us to go with this approach would be that we can't prevent `PostUploadHandler` to handle the `OnPostUploaded` event. I mean, we can, but that'd be very complicated. So, I think we at least want push post action to be handled by `PostUploadHandler` and might even prefer to do the `autosave` there as well.

Whatever we do in the use case, it'll have a result (success or error) so as long as we handle them correctly, I don't think we can ever be in an inconsistent state.

---

@malinajirka Thank you for sharing your thoughts!

Similar to our previous Slack conversation, I think one or both of us are missing something which makes it hard to communicate effectively. Let's tackle this together on the screenshare session on Monday.